### ISSUE TITLE
Unfair training process

### ISSUE THREAD

**Issue by LoveMiki** [arjish/PreTrainedFullLibrary_FewShot#2]
The library-based methods pre-trained on the whole ILSVRC dataset and validated on the CUB dataset (1,200 classes were actually used), while the meta-learning based methods trained and validated only on the ILSVRC dataset (1,000 classes).  For a fair comparison, why didn't you just use the pre-trained bockbones for the meta-learning methods?

**Comment by arjish** [comment]
Point 1: 
The implementations of meta-learning methods need a validation set. Since we want a cross-domain application, the training should solely happen on ILSVRC. Hence for the meta-learning methods, we split ILSVRC into 900 training and 100 validation classes. Now, if we remove the 100 classes and give all 1000 classes for training, then the meta-learner will have no validation set and one can argue it to be a reason for underperforming since the original implementations used a validation set to improve training. 

Point 2:
Using pre-trained backbones means resorting to transfer learning which we did for transfer-learning based methods (FEAT,
Meta-transfer, and SUR). Other meta-learning based methods we compared with do not use pre-trained backbones.

Hope this makes the design decision clear and fair.

**Comment by LoveMiki** [comment]
> Point 1: The implementations of meta-learning methods need a validation set. Since we want a cross-domain application, the training should solely happen on ILSVRC. Hence for the meta-learning methods, we split ILSVRC into 900 training and 100 validation classes. Now, if we remove the 100 classes and give all 1000 classes for training, then the meta-learner will have no validation set and one can argue it to be a reason for underperforming since the original implementations used a validation set to improve training.
> 
> Point 2: Using pre-trained backbones means resorting to transfer learning which we did for transfer-learning based methods (FEAT, Meta-transfer, and SUR). Other meta-learning based methods we compared with do not use pre-trained backbones.
> 
> Hope this makes the design decision clear and fair.

I know that we need the validation dataset to select the network with the best parameters. However, when your library-based methods pre-trained on the whole ILSVRC dataset and validated on the CUB dataset, it actually determines the best parameters for the CUB dataset (cross-domain validation), while meta-learning methods still select the best parameters for only 100 classes from the ILSVRC dataset.
